### PR TITLE
Implement multiline input

### LINKED ISSUE #4: Activate parsing of multiline strings
Currently there's an ugly hack of using an `<input>` element (which only supports a single line) because I wasn't able to pass a multiline string from the `<textarea>` to docutils's [publish_string()](https://docutils.sourceforge.io/docs/api/publisher.html):

https://github.com/waldyrious/rst-playground/blob/f8698e0d0c33e1bfe324e1e126884e388bacf6d3/index.html#L71-L72

This means the demo is only useful for single-line rST input, which is clearly inadequate. A proper fix should be implemented to allow using multiline strings and get rid of the `<input>` hack.

### PR DESCRIPTION
**Pull request by not-my-profile**
Fixes #4, ~~fixes #7~~ (moved to #12), ~~fixes #8~~ (moved to #10) ~~and other stuff~~ (moved to #9 and #13).

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+}
+let pyodideReadyPromise = main();
+
+async function rstToHtml() {
+  let pyodide = await pyodideReadyPromise;
+  try {
+    // TODO: also load pygments for syntax highlighting
+    await pyodide.loadPackage("docutils");
+
+    pyodide.globals.set('input_text', input.value);
```
Comment (waldyrious) [inline script.js:28]: Awesome! Could you add a comment + link to explain why this is needed?

**Comment by waldyrious** [comment]
Thanks so much for tackling these! Would you mind moving 713ef32642db1386dcd7cdce551020d3338653ae to a separate PR? I'd like to discuss those changes separately. And maybe also split out the moving of the CSS and JS code to separate files, so that this PR's global diff is more readable.

I can do both if you'd prefer.

**Comment by waldyrious** [comment]
Also, 5c1055b7b814215e3b2329152fdca8d0d593ddfe is already included in #1. Let me merge that one first since these changes are more substantial (though fortunately they don't overlap much).

File: script.js (hunk 1 above)
Comment (not-my-profile) [reply script.js:28]: What do you mean? I think it's quite obvious why this is needed.

**Comment by waldyrious** [comment]
No worries, it's just my OCD :sweat_smile: Also, GitHub's per-commit view is a little cumbersome to use. I'll split the PR.

File: script.js (hunk 1 above)
Comment (waldyrious) [reply script.js:28]: I meant as opposed to the previous approach of interpolating the input content into the JS [template literal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) defining the Python code, which would be the (IMHO) intuitive approach.

**Comment by waldyrious** [comment]
Update: I've split the other changes into #9 and #10, and rebased this PR (dropping 5c1055b7b in favor of the changes in #1). Sorry for the inconvenience :)

I still would like to split this further, but I'm out of time for the moment, and will pick this up later.

File: script.js (hunk 1 above)
Comment (not-my-profile) [reply script.js:28]: Done :)

Comment (waldyrious) [reply script.js:27]: Following our discussion on Matrix, it became evident that properly explaining why the earlier approach didn't work would be too verbose for a code comment. I decided to go with a commit message instead, which I'll add as a comment below for convenience as well.

**Comment by waldyrious** [comment]
_(Adding the expanded commit message with the explanation here, for convenience)_

The previous approach of interpolating the contents of the textarea
directly into the JavaScript template literal containing Python code
didn't work because:
1. The string contained newlines,
   so it had to be specified using triple quotes (""").
2. The template literal is indented within the JavaScript code,
   and the whitespace characters used for indentation
   are part of the string; i.e. the whole Python script is indented;
   however, when the textarea contents got interpolated into it,
   the resulting Python code would contain negative indentation,
   something like this:
   ```python
       from docutils.core import publish_string
       publish_string("""foo
   bar""", writer_name='html5').decode("utf-8")
   ```
   ...so Python would complain about invalid syntax.

Passing the string via a variable
(rather than injecting it directly into the Python code)
avoids this issue because the Python code doesn't change shape
depending on the input text.

In any case, it's unadvisable to directly interpolate user input into code,
so the new approach is also safer.

**Comment by waldyrious** [comment]
WDYT of the PR as it stands, @not-my-profile?

**Pull request merged**